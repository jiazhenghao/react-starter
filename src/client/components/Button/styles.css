.button_style {
  border: 1px solid green;
  padding: 5px;
}
