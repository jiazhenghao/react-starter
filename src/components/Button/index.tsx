import React from 'react'
import './styles.css'

interface ButtonProps {
  label: string
}

const Button: React.FC<ButtonProps> = ({ label }) => {
  return (
    <div data-testid="button" className="button_style">
      {label}
    </div>
  )
}

export default Button
