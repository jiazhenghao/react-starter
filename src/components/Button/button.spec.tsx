/**
 * @jest-environment jsdom
 */
import React from 'react'
import ReactDOM from 'react-dom'
import Button from '@/components/Button/'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)

test('it renders without crash', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Button label="please click" />, div)
})

test('it renders button well', () => {
  const { getByTestId } = render(<Button label="please click on me" />)
  expect(getByTestId('button')).toHaveTextContent('please click on me')
})

test('it renders button well', () => {
  const { getByTestId } = render(<Button label="save" />)
  expect(getByTestId('button')).toHaveTextContent('save')
})

test('it matches snapshot', () => {
  const tree = renderer.create(<Button label="save" />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('it matches another snapshot', () => {
  const tree = renderer.create(<Button label="quit and go" />).toJSON()
  expect(tree).toMatchSnapshot()
})
