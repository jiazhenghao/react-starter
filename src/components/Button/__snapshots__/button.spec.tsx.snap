// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches another snapshot 1`] = `
<div
  className="button_style"
  data-testid="button"
>
  quit and go
</div>
`;

exports[`it matches snapshot 1`] = `
<div
  className="button_style"
  data-testid="button"
>
  save
</div>
`;
